services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,58 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+  if(event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  } else if(thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  } else {
+    BaseType_t errCode = xQueueSend(thermalMgrQueueHandle, event, 0);
+    if(errCode != pdTRUE) {
+      return ERR_CODE_QUEUE_FULL;
+    }
+    return ERR_CODE_SUCCESS;
+  }
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t data = {0};
+
+    BaseType_t errCodeEvent = xQueueReceive(thermalMgrQueueHandle, &data, portMAX_DELAY);
+
+    if(errCodeEvent == pdTRUE) {
```

Comment:
nit: in general if you arent going to use the errCode anywhere outside of the if condition, its better to just call the function in the if condition and compare it there rather than using another variable. Like in the comment shown above